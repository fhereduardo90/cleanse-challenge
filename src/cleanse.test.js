const cleanse = require('./cleanse')
const mocks = require('../mocks')

describe('cleanse', () => {
  it('should remove properties with values null or undefined', () => {
    const firstExampleResult = cleanse(mocks.firstExample)
    const secondExampleResult = cleanse(mocks.secondExample)

    expect(firstExampleResult).toMatchSnapshot()
    expect(secondExampleResult).toMatchSnapshot()
    expect(secondExampleResult).not.toHaveProperty('nickname')
    expect(secondExampleResult).not.toHaveProperty('address')
    expect(secondExampleResult).toHaveProperty('active', false)
    expect(secondExampleResult.friends[0]).toHaveProperty('active', true)
    expect(secondExampleResult.friends[0]).not.toHaveProperty('address')
  })
})