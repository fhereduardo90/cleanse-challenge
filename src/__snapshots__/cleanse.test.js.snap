// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cleanse should remove properties with values null or undefined 1`] = `
Object {
  "firstName": "Peter",
  "friends": Array [
    Object {
      "age": 0,
      "lastName": "Banner",
      "nickname": "hulk",
    },
    Object {
      "address": false,
      "firstName": "Tony",
      "friends": Array [
        Object {
          "friends": Array [],
          "nickname": "war machine",
        },
      ],
      "lastName": "Stark",
      "nickname": "iron man",
    },
  ],
  "lastName": "Parker",
  "nickname": "spiderman",
}
`;

exports[`cleanse should remove properties with values null or undefined 2`] = `
Object {
  "active": false,
  "firstName": "Black",
  "friends": Array [
    Object {
      "active": true,
      "firstName": "Natasha",
      "lastName": "Romanova",
      "nickname": "Black Widow",
    },
  ],
  "lastName": "Panter",
}
`;
